### DISCUSSION TITLE
Update index instead of deleting and adding

### DISCUSSION THREAD

**Discussion by msc2402** [OrchardCMS/OrchardCore#10916] [Q&A]
Hi,

I was wondering if there is any way to update an existing MapIndex instead of deleting it and adding it.
I'm currently facing the issue that the index table has an Integer Id with a limited max value. 

I'm keeping track of certain crypto coins and storing them in the database. Using a websocket I'm receiving the latest prices continually of a specific coin which I will then save to the database. Currently my ID of the index table increments rather quickly, as I'm updating the Document record frequently. 

What I fear is that I will reach the limitations of the int ID quickly this way.

TLDR:
Is there any way to let an IndexProvider update an existing MapIndex record, instead of deleting + adding.

````
namespace Core.Indexes
{
    public class CandleIndex : MapIndex
    {
        public string Symbol { get; set; }
        public TimeInterval Interval { get; set; }

        public DateTime StartDate { get; set; }
    }

    public class CandleIndexProvider : IndexProvider<Candle>
    {
        public override void Describe(DescribeContext<Candle> context)
        {
            context.For<CandleIndex>()
                .Map(candle =>
                {
                    return new CandleIndex
                    {
                        Symbol = candle.Symbol,
                        Interval = candle.Interval,
                        StartDate = candle.StartDate
                    };
                });
        }
    }
}
````

The ID distance between yesterday's price candle index and today:
![image](https://user-images.githubusercontent.com/4236179/147139495-5519436f-8ade-446a-be96-3199e6340ea9.png)

**Comment by Skrypt** [comment]
Good one but I believe this is a question to raise in the YesSQL repository.

**Reply by msc2402** [reply]
You're probably right, thanks.
